src/_P101_WakeOnLan.ino

```diff
@@ -0,0 +1,475 @@
+//   PC's with integrated WiFi and Ethernet ports will likely have two MAC addresses, with specific
+//   WOL settings for each one.
+//   I don't believe USB connected network adapters are capable of WOL; Check the adapter's manual.
+//
+// Useful WOL tools:
+// WakeMeOnLan WOL Scanner/Sender: https://www.nirsoft.net/utils/wake_on_lan.html
+// wolsniffer WOL Packet Analyzer: https://apreltech.com/Downloads/wolsniffer.zip
+// Online MAC Address Generator  : https://www.browserling.com/tools/random-mac
+//
+// ************************************************************************************************
+
+#include "_Plugin_Helper.h"
+#include <WakeOnLan.h>
+
+// Plugin defines
+#define PLUGIN_101
+#define PLUGIN_ID_101      101
+#define PLUGIN_NAME_101    "Communication - Wake On LAN [Testing]"
+
+// Config Setting defines
+#define CUSTOMTASK_STR_SIZE_P101 20
+#define DEF_TASK_NAME_P101 "WAKE_ON_LAN"
+#define UDP_PORT_P101      ExtraTaskSettings.TaskDevicePluginConfigLong[1]
+#define FORM_PORT_P101     "P101_port"
+
+// Command keyword defines
+#define CMD_NAME_P101      "WAKEONLAN"
+
+// MAC Defines.
+#define MAC_ADDR_SIZE_P101 17  // MAC Addr String size (fixed length). e.g. FA:39:09:67:89:AB
+#define MAC_BUFF_SIZE_P101 18  // MAC Addr Buffer size, including NULL terminator.
+#define MAC_SEP_CHAR_P101  ':' // MAC Addr segments are separated by a colon.
+#define MAC_SEP_CNT_P101   5   // MAC Addr colon separator count for valid address.
+#define MAC_STR_DEF_P101   "00:00:00:00:00:00"
+#define MAC_STR_EXP_P101   "5d:89:22:56:9c:60"
+
+// IP Defines.
+#define IP_ADDR_SIZE_P101  15  // IPv4 Addr String size (max length). e.g. 192.168.001.255
+#define IP_BUFF_SIZE_P101  16  // IPv4 Addr Buffer size, including NULL terminator.
+#define IP_MIN_SIZE_P101   7   // IPv4 Addr Minimum size, allows IP strings as short as 0.0.0.0
+#define IP_SEP_CHAR_P101   '.' // IPv4 Addr segments are separated by a dot.
+#define IP_SEP_CNT_P101    3   // IPv4 Addr dot separator count for valid address.
+#define IP_STR_DEF_P101    "255.255.255.255"
+
+// Port Defines.
+#define PORT_DEF_P101      9
+#define PORT_MAX_P101      65535
+
+// Misc Defines
+#define LOG_NAME_P101      "WAKE ON LAN: "
+
+// ************************************************************************************************
+// WOL Objects
+WiFiUDP   udp;
+WakeOnLan WOL(udp);
+
+// ************************************************************************************************
+// Local Functions
+bool validateIp(String ipStr);
+bool validateMac(String macStr);
+bool validatePort(String portStr);
+
+// ************************************************************************************************
+
+boolean Plugin_101(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  char   ipString[IP_BUFF_SIZE_P101]   = "";
+  char   macString[MAC_BUFF_SIZE_P101] = "";
+  String msgStr;
+  const String wolStr = F(LOG_NAME_P101);
```

Comment:
Please try to allocate those only when needed, otherwise they are also created each time the plugin is called like in `PLUGIN_FIFTY_PER_SECOND` even though this plugin doesn't even handle such a case.

Reply:
I'll correct this.
- Thomas

Reply:
Update file. Allocations now done in local scope for PLUGIN_WEBFORM_LOAD, PLUGIN_WEBFORM_SAVE, and PLUGIN_WRITE.
- Thomas